implementations/rust/ockam/ockam_app/src/projects/commands.rs

```diff
@@ -13,6 +13,32 @@ type SyncState = Arc<RwLock<ProjectState>>;
 
 // At time of writing, tauri::command requires pub not pub(crate)
 
+#[tauri::command]
+pub async fn create_enrollment_ticket<R: Runtime>(
+    project_id: String,
+    app: AppHandle<R>,
+) -> Result<EnrollmentTicket, String> {
+    let state: State<'_, SyncState> = app.state();
+    let reader = state.read().await;
+    let project: &Project = reader
+        .iter()
+        .find(|p| p.id == project_id)
+        .ok_or_else(|| format!("Project {project_id} not found"))?;
+
+    // TODO:
+    // - retrieve or create TrustContext
+    // - retrieve identity?
+    // - create secure channel to authority
+    // - calculate route for DefaultAddress::ENROLLMENT_TOKEN_ISSUER
+    // - instantiate TokenIssuerClient for route
+    // - create token with attributes and TTL
+    // - convert token to EnrollmentTicket with project and trust context
+    //
+    // see also: ockam_command::project::ticket
```

Comment:
Perhaps it would be easier to call the command directly. Here are the two approaches:

The hard way, you would have to implement some missing methods to make this work:

```rust
let cli_state = state.state().await;
let node = cli_state.nodes.get(NODE_NAME)?.config();
let identity_name = node.identity_name()?; // Missing
let project = cli_state.projects.get(project_name)?;
let project_lookup = ProjectLookup::from_project(&project.config()).await?;
let node_manager = state.node_manager_worker().await;
let sc = node_manager.create_secure_channel().await?; // Missing
let token_issuer_route = ..;
let client = TokenIssuerClient::new(..);
let token = client.create_token..;
let ticket = EnrollmentTicket::new(token, project, trust_context);
```

The easier way (if `project_name` is always going to be `"default"` you could even ignore the last two args):
```rust
let bin = get_env_with_default("OCKAM", "ockam".to_string())?;
let hex_encoded_ticket = duct::cmd!(
    bin,
    "project",
    "ticket",
    "--project",
    project_name,
    "--to",
    &format!("/project/{project_name}")
)
.read()?;
let ticket: EnrollmentTicket = serde_json::from_slice(&hex::decode(hex_encoded_ticket)?)?;
```

Reply:
I went with an approach based on your second snippet for now - I'd love to be able to do this operation in-process in the future.

Reply:
I've created an issue so we don't forget about it: https://github.com/build-trust/ockam/issues/5653